### PR TITLE
Update supported Swift/Xcode Versions.

### PR DESCRIPTION
**Pull request by thomasvl**
Update the documentation to reflect that we will try to support the older of the Swift NIO supported version and the minimum Xcode version supported for the Apple AppStore.

As of today, that means:

- Swift NIO: Swift 5.6 - https://github.com/apple/swift-nio#swiftnio-2
- AppStore: Xcode 14.1 - https://developer.apple.com/news/?id=jd9wcyov#:~:text=Starting%20April%2025%2C%202023%2C%20iOS,or%20later.

Xcode 14.1 included Swift 5.7. So we'll support Swift 5.6+, but the Apple deployment minimums are moved up to the values from Xcode 14.1.

In addition to the documentation updates, this commit also:

- Updates the main Package.swift and SwiftProtobuf.podspec for the new deployment minimums.
- Updates the CI setup to the minimum Swift version.

NOTE: There are no code changes in this commit, once this CL is landed, the code changes will be done in follow ups since there are generation changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,10 +90,9 @@ your project as explained below.
 
 To use Swift with Protocol buffers, you'll need:
 
-* A Swift 5.0 or later compiler (Xcode 10.2 or later).  Support is included for
-  the Swift Package Manager. The Swift protobuf project is being developed and
-  tested against the latest release version of Swift available from
-  [Swift.org](https://swift.org)
+* A Swift 5.6 or later compiler (Xcode 14.1 or later). The Swift protobuf project
```
Comment (allevato) [inline README.md:93]: When I read "Swift x.y (Xcode x.y)", I tend to interpret those as being associated together, but that's not the case here since the minimum Xcode has a higher version than the overall Swift version we support, due to Apple factors.

Maybe rephrase this to be more explicit about the divergence, like this?

> A Swift 5.6 or later compiler (or, if building with Xcode, Xcode 14.1 or later as required by the App Store)

If there's a future update where the minimum NIO supported version is equal to or greater than the minimum supported Xcode version, then we can condense it back down to "A Swift 5.6 or later compiler (Xcode x.y)".

Comment (thomasvl) [reply README.md]: Done.

**Review by allevato — APPROVED** [review]

File: Package.swift [hunk 2]
```diff
@@ -13,6 +13,12 @@ import PackageDescription
 
 let package = Package(
   name: "SwiftProtobuf",
+  platforms: [
+      .iOS(.v10),
```
[the remark this answers is not in the document]
Comment (thomasvl) [reply Package.swift:17]: Yup. I don't think it's worth messing with the 1.x branch at the moment, I'm hoping we can sunset it in the near future.

Should I tweak the one `.md` to call out if the new Xcode raises a deployment minimum that that forces a major version change on the library? Just so we always remember that.

[the remark this answers is not in the document]
Comment (FranzBusch) [reply Package.swift:17]: I don't think we want to do this. I think we should just remove the part about the minion deployment version and only consider the Swift version.

[the remark this answers is not in the document]
Comment (thomasvl) [reply Package.swift:17]: I don't think we can leave it out of the `.podspec`, but I'll drop it here.

[the remark this answers is not in the document]
Comment (thomasvl) [reply Package.swift]: Updated.

[the remark this answers is not in the document]
Comment (thomasvl) [reply Package.swift]: > I don't think we can leave it out of the `.podspec`, but I'll drop it here.

I tried removing the deployment attributes from the `.podspec`, and `pod lib lint` fails, from the errors, CocoaPods must use some hard coded values vs. getting some defaults from Xcode (atleast I'd hope Xcode didn't try to default to macOS 10.6 and iOS 8.0 since Xcode 14.3.1 can't really build for those). So it looks like we do need to keep the values in the `.podspec`.

**Review by tbkka — APPROVED** [review]
Looks good!
